pkg/diagnostics/networkpod/external.go

```diff
@@ -0,0 +1,44 @@
+package network
+
+import (
+	"fmt"
+
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+const (
+	CheckExternalNetworkName = "CheckExternalNetwork"
+)
+
+// CheckExternalNetwork is a Diagnostic to check accessibility of external network within a pod
+type CheckExternalNetwork struct {
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d CheckExternalNetwork) Name() string {
+	return CheckExternalNetworkName
+}
+
+// Description is part of the Diagnostic interface and just returns the diagnostic description.
+func (d CheckExternalNetwork) Description() string {
+	return "Check that external network is accessible within a pod"
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d CheckExternalNetwork) CanRun() (bool, error) {
+	return true, nil
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d CheckExternalNetwork) Check() types.DiagnosticResult {
+	r := types.NewDiagnosticResult(CheckExternalNetworkName)
+
+	externalAddress := "www.google.com"
```

Comment:
I'd commented before that RH IT might not want us using "www.redhat.com", but we *definitely* shouldn't be using *someone else's* server here. (Using "www.redhat.com" would be better.)

Also, I'm not sure of the context of the diagnostic tests and how we interpret/present the results, but we need to keep in mind that the cluster may intentionally be firewalled off from the internet, so if this test fails it doesn't necessarily mean that something is wrong; it's just something for the admin to be made aware of in case the failure *wasn't* expected.